From 39d26115250c9c233018e487ce495e5a16faacc2 Mon Sep 17 00:00:00 2001
From: Ludwig Nussel <ludwig.nussel@suse.de>
Date: Wed, 3 Jul 2013 11:25:38 +0200
Subject: [PATCH] Don't require Mozilla::CA

IO::Socket::SSL will fall back to using the system's root
CA-certificates if no options are passed.
---
 Makefile.PL               |  1 -
 README                    |  9 +--------
 lib/LWP/Protocol/https.pm | 23 -----------------------
 3 files changed, 1 insertion(+), 32 deletions(-)

Index: LWP-Protocol-https-6.07/Makefile.PL
===================================================================
--- LWP-Protocol-https-6.07.orig/Makefile.PL
+++ LWP-Protocol-https-6.07/Makefile.PL
@@ -26,7 +26,6 @@ my %WriteMakefileArgs = (
                     'LWP::UserAgent' => '6.06',
                     'Net::HTTPS' => 6,
                     'IO::Socket::SSL' => "1.54",
-                    'Mozilla::CA' => "20110101",
                     'perl' => '5.008001',
                 },
             },
Index: LWP-Protocol-https-6.07/README
===================================================================
--- LWP-Protocol-https-6.07.orig/README
+++ LWP-Protocol-https-6.07/README
@@ -17,13 +17,6 @@ DESCRIPTION
     you don't use it directly. Once the module is installed LWP is able to
     access sites using HTTP over SSL/TLS.
 
-    If hostname verification is requested by LWP::UserAgent's "ssl_opts",
-    and neither "SSL_ca_file" nor "SSL_ca_path" is set, then "SSL_ca_file"
-    is implied to be the one provided by Mozilla::CA. If the Mozilla::CA
-    module isn't available SSL requests will fail. Either install this
-    module, set up an alternative "SSL_ca_file" or disable hostname
-    verification.
-
     This module used to be bundled with the libwww-perl, but it was
     unbundled in v6.02 in order to be able to declare its dependencies
     properly for the CPAN tool-chain. Applications that need https support
@@ -31,7 +24,7 @@ DESCRIPTION
     longer need to know what underlying modules to install.
 
 SEE ALSO
-    IO::Socket::SSL, Crypt::SSLeay, Mozilla::CA
+    IO::Socket::SSL, Crypt::SSLeay
 
 COPYRIGHT
     Copyright 1997-2011 Gisle Aas.
Index: LWP-Protocol-https-6.07/lib/LWP/Protocol/https.pm
===================================================================
--- LWP-Protocol-https-6.07.orig/lib/LWP/Protocol/https.pm
+++ LWP-Protocol-https-6.07/lib/LWP/Protocol/https.pm
@@ -23,7 +23,8 @@ sub _extra_sock_opts
     else {
 	$ssl_opts{SSL_verify_mode} = 0;
     }
-    if ($ssl_opts{SSL_verify_mode}) {
+    my $suse_allows_mozilla_ca = 0;
+    if ($suse_allows_mozilla_ca && $ssl_opts{SSL_verify_mode}) {
 	unless (exists $ssl_opts{SSL_ca_file} || exists $ssl_opts{SSL_ca_path}) {
 	    eval {
 		require Mozilla::CA;
